clientcore/mixClient_test.go

```diff
@@ -36,33 +35,33 @@ var mixes []config.MixConfig
 
 func Setup() error {
 	for i := 0; i < 10; i++ {
-		pub, _, err := sphinx.GenerateKeyPair()
+		_, pub, err := sphinx.GenerateKeyPair()
 		if err != nil {
 			return err
 		}
-		mixes = append(mixes, config.NewMixConfig(fmt.Sprintf("Mix%d", i), "localhost", strconv.Itoa(3330+i), pub))
+		mixes = append(mixes, config.NewMixConfig(fmt.Sprintf("Mix%d", i), "localhost", strconv.Itoa(3330+i), pub.Bytes()))
 	}
 
 	// Create a mixClient
-	pubC, privC, err := sphinx.GenerateKeyPair()
+	privC, pubC, err := sphinx.GenerateKeyPair()
 	if err != nil {
 		return err
 	}
-	client = NewCryptoClient(pubC, privC, elliptic.P224(), config.MixConfig{}, NetworkPKI{})
+	client = NewCryptoClient(privC, pubC, config.MixConfig{}, NetworkPKI{})
 
 	//Client a pair of mix configs, a single provider and a recipient
-	pub1, _, err := sphinx.GenerateKeyPair()
+	_, pub1, err := sphinx.GenerateKeyPair()
 	if err != nil {
 		return err
 	}
 
-	pub2, _, err := sphinx.GenerateKeyPair()
+	_, pub2, err := sphinx.GenerateKeyPair()
 	if err != nil {
 		return err
 	}
 
-	m1 := config.MixConfig{Id: "Mix1", Host: "localhost", Port: "3330", PubKey: pub1}
-	m2 := config.MixConfig{Id: "Mix2", Host: "localhost", Port: "3331", PubKey: pub2}
+	m1 := config.MixConfig{Id: "Mix1", Host: "localhost", Port: "3330", PubKey: pub1.Bytes()}
+	m2 := config.MixConfig{Id: "Mix2", Host: "localhost", Port: "3331", PubKey: pub2.Bytes()}
```

Comment:
Is it worth later firming up `PubKey` in the `MixConfig` so that it is strongly typed rather than `byte[]`?

Reply:
I'm not sure how easy (or not) that is going to be as currently the keys are defined as constant size byte arrays as defined by the curve used. Meanwhile the configs are defined in protobuf files and protobuf does not allow using constant size arrays.